internal/cluster/distribution/pke/pkeaws/pkeawsworkflow/activity_nlb_health_check.go

```diff
@@ -0,0 +1,143 @@
+// NewHealthCheckActivity returns a new HealthCheckActivity.
+func NewHealthCheckActivity(
+	awsSessionFactory AWSSessionFactory,
+	elbv2Factory ELBV2APIFactory,
+) *HealthCheckActivity {
+	return &HealthCheckActivity{
+		awsSessionFactory: awsSessionFactory,
+		elbv2Factory:      elbv2Factory,
+	}
+}
+
+type HealthCheckActivityInput struct {
+	OrganizationID uint
+	SecretID       string
+	Region         string
+	ClusterID      uint
+	ClusterName    string
+}
+
+// HealthCheckActivityOutput holds the output data
+type HealthCheckActivityOutput struct {
+	nlbStatus []*elbv2.TargetHealthDescription
+}
+
+// Register registers the activity in the worker.
+func (a *HealthCheckActivity) Register() {
+	activity.RegisterWithOptions(a.Execute, activity.RegisterOptions{Name: HealthCheckActivityName})
+}
+
+func (a *HealthCheckActivity) Execute(ctx context.Context, input HealthCheckActivityInput) (*HealthCheckActivityOutput, error) {
+	if a == nil {
+		return nil, errors.New("activity is nil")
+	}
+	session, err := a.awsSessionFactory.New(input.OrganizationID, input.SecretID, input.Region)
+	if err != nil {
+		return nil, errors.WrapIf(err, "failed to create AWS session")
+	}
+
+	elbv2Svc := a.elbv2Factory.New(session)
+	nlbName := "pke-" + input.ClusterName + "-nlb"
+
+	backoffConfig := backoff.ConstantBackoffConfig{
+		Delay:      backoffDelay,
+		MaxRetries: backoffMaxRetries,
+	}
+	backoffPolicy := backoff.NewConstantBackoffPolicy(backoffConfig)
+
+	var loadBalancerOutput *elbv2.DescribeLoadBalancersOutput
+	if err := backoff.Retry(func() error {
+		var errInt error
+		loadBalancerOutput, errInt = elbv2Svc.DescribeLoadBalancers(
+			&elbv2.DescribeLoadBalancersInput{
+				Names: []*string{&nlbName},
+			})
+		if errInt != nil {
+			return errors.WrapIf(errInt, "failed to describe load balancers")
+		}
+
+		return nil
+	}, backoffPolicy); err != nil {
+		return nil, err
+	}
+
+	var targetGroupOutput *elbv2.DescribeTargetGroupsOutput
+	if err := backoff.Retry(func() error {
+		var errInt error
+		targetGroupOutput, errInt = elbv2Svc.DescribeTargetGroups(
+			&elbv2.DescribeTargetGroupsInput{
+				LoadBalancerArn: loadBalancerOutput.LoadBalancers[0].LoadBalancerArn,
+			})
+		if errInt != nil {
+			return errors.WrapIf(errInt, "failed to describe target groups")
+		}
+
+		return nil
+	}, backoffPolicy); err != nil {
+		return nil, err
+	}
+
+	var targetHealthOutput *elbv2.DescribeTargetHealthOutput
+	if err := backoff.Retry(func() error {
+		var errInt error
+		targetHealthOutput, errInt = elbv2Svc.DescribeTargetHealth(
+			&elbv2.DescribeTargetHealthInput{
+				TargetGroupArn: targetGroupOutput.TargetGroups[0].TargetGroupArn,
+			})
+		if errInt != nil {
+			return errors.WrapIf(errInt, "failed to describe target health")
+		}
+
+		return nil
+	}, backoffPolicy); err != nil {
+		return nil, err
+	}
+
+	for _, tgHealhtDesc := range targetHealthOutput.TargetHealthDescriptions {
+		if *tgHealhtDesc.TargetHealth.State != elbv2.TargetHealthStateEnumHealthy {
```

Comment:
**Recommended**: we **SHOULD** add a preceding `lgHealthDesc.TargetHealth.State != nil` condition to make sure we are not trying to dereference a nil pointer.

Reply:
fixed: https://github.com/banzaicloud/pipeline/pull/3229/commits/6146e54ad33a4aa0d22fad813f9c67841de394ea